### ISSUE TITLE
pm: Fetch OrchestratorInfo updates for a broadcast session in response

### ISSUE THREAD

**Issue by yondonfu** [livepeer/go-livepeer#875]
When a broadcaster creates a broadcast session for a particular orchestrator, it should subscribe to OrchestratorInfo updates from the orchestrator so that the broadcaster can be notified whenever the orchestrator updates the required ticket params (when we implement payment accounting this subscription can also be used to receive notifications of updated price).

The subscription would be implemented as a service side streaming RPC call

```
service Orchestrator {
    ...

    // Called by the broadcaster to receive streaming orchestrator info updates
    rpc SubscribeOrchestratorUpdates(OrchestratorRequest) returns (stream OrchestratorInfo)
}
```

```
func (h *lphttp) SubscribeOrchestratorUpdates(..., stream net.Orchestrator_SubscribeOrchestratorUpdatesServer) error {
    // gasPriceNotifier is a global instance of an object that continuous watches for network
    // gas price updates. The object can be used to subscribe to updates via a channel
    sub := h.gasPriceNotifier.Subscribe()

    defer sub.Cancel()
    
    for {
        case gasPrice := <-sub.Updates():
            // Update ticket params
            ticketParams := TicketParams(gasPrice)
            // With payment accounting, update price if needed as well
            info := &net.OrchestratorInfo{
                ...
                ticketParams,
            }

            if err := stream.Send(info); err != nil {
                return err
            }
        
        // Case to detect server shut down
    }
}
```

**Comment by j0sh** [comment]
Subscribing via a separate RPC call is one way to do it.

Another is to return the updated pricing info as part of a segment submission request - submit a segment, receive an error response indicating a price update, and go from there. That makes use of the mechanisms we already have in place, rather than adding another RPC call which expands the surface area of the networking protocol.

**Comment by yondonfu** [comment]
Yeah I see the benefit in minimizing networking surface area. If O responds with an `OrchestratorInfo` message (or whatever message includes ticket params + pricing) in the error I guess B could check the HTTP status code (`http.StatusPaymentRequired`?) to determine whether it needs to parse the error for the ticket params + pricing and any other relevant orchestrator info?

**Comment by j0sh** [comment]
> I guess B could check the HTTP status code (http.StatusPaymentRequired?) to determine whether it needs to parse the error for the ticket params + pricing

We could also put in additional structured error types to the protobuf `TranscodeResult` 

https://github.com/livepeer/go-livepeer/blob/d0060bd79fb0e2f60767bc72988e1c2ec10ca45d/net/lp_rpc.proto#L139-L143

Maybe something like a field `OrchestratorInfo info = 16`  [1].

This approach is also nice in that it allows us to update other elements of `OrchestratorInfo` such as transcoder URI, allowing the O to redirect its own load. Embedding this in the segment reply means less set-up and tear-down for each new O the broadcaster works with. We do need to be mindful of things like O-induced forwarding loops or "oscillating updates" but that's something we can address during implementation by capping the # of hops.

[1] Using  field number of 16 here because larger field numbers take up more space and I expect this particular error will be infrequent so we can reserve smaller numbers for fields that are more likely to occur; see https://developers.google.com/protocol-buffers/docs/proto3#assigning-field-numbers

**Comment by yondonfu** [comment]
Closed by #905
